docs/metrics.md

```diff
@@ -0,0 +1,422 @@
+
+
+##### `addons`
+
+A list of IDs of installed add-ons. Array.
+
+
+##### `adBlocker`
+
+Does the user have an add-on installed that is known to block ads? Boolean.
+
+
+##### `openTabs`
+
+The count of open tabs across all windows. Number.
+
+
+##### `openWindows`
+
+The number of open windows across the browser. Number.
+
+
+##### `hangs`
+
+The number of recent hangs across the browser. Number.
+
+
+#### Page data
+
+Data about the page from which the report was initiated.
+
+
+##### `timestamp`
+
+The timestamp from which the page request began. Integer.
+
+
+##### `protocol`
+
+The protocol used to retrieve the page, as reported by `window.location.protocol`. String.
+
+
+##### `protocolVersion`
+
+The version of the protocol used to retrieve the page, as reported by the `Version` header. String.
+
+
+##### `trackingProtection`
+
+Is tracking protection enabled for the page? Boolean.
+
+
+##### `timerContentLoaded`
+
+Number of milliseconds between the start of the page load and the `DOMContentLoaded` event. Number.
+
+
+##### `timerWindowLoad`
+
+Number of milliseconds between the start of the page load and the `load` event on `window`. Number.
+
+
+##### `timerFirstPaint`
+
+Number of milliseconds between the start of the page load and the first non-blank paint. Number.
+
+
+##### `timerFirstInteraction`
+
+Number of milliseconds between the start of the page load and the first of the following events:
+
+* `click`
+* `touch`
+* `scroll`
+* `keypress`
+
+Number. Will return `null` if none of these events took place.
+
+
+##### `requests`
+
+An object containing information about requests spawned by the page. Each member key indicates the type of request:
+
+* `html`: HTML files
+* `css`: Cascading style sheets
+* `js`: JavaScript
+* `xhr`: files retrieved by `XMLHttpRequest`
+* `fonts`: non-system fonts
+* `images`: image files
+* `media`: audio and video files
+* `flash`: Flash embeds
+* `ws`: WebSocket connections
+* `other`: anything not covered in the above categories
+* `all`: all requests summed.
+
+These are categorized in the same way as the [Firefox Developer Tools Network Monitor](https://developer.mozilla.org/docs/Tools/Network_Monitor).
+
+Each value of the object contains three data points:
+
+* `num`: the number of requests of that type
+* `size`: the summed filesize of requests of that type, in bytes
+* `time`: the time it took to load those filesge. Each member ke, in ty indicates the type of request:
+
+
+##### `externalRequests`
+
+A count of requests made to different hostnames. Number.
+
+
+##### `disconnectRequests`
+
+A count of requests made to hosts on the [disconnect.me tracking protection blocklist](https://disconnect.me/trackerprotection). Number.
+
+
+##### `consoleErrors`
+
+A count of the errors logged to the browser console. Number.
+
+
+##### `amountCached`
+
+The proportion of requests returning `304` status codes. Number, between `0.0` and `1.0`.
+
+
+##### `engagement`
```

Comment:
I have it broken out for both:
- https://github.com/chuckharmston/pulse/blob/ce5ad5c40425d3dac78bf67918edd26b7c243942/docs/metrics.md#engagement
- https://github.com/chuckharmston/pulse/blob/ce5ad5c40425d3dac78bf67918edd26b7c243942/docs/metrics.md#sessionengagement